src/typeclass_stats.lean

```diff
@@ -0,0 +1,122 @@
+import algebra.group.defs
+
+open tactic declaration environment native
+
+meta def pos_line (p : option pos) : string :=
+match p with
+| some x := to_string x.line
+| _      := "null"
+end
+
+def prepare_filename₁ : string → list string :=
+λ xs, ((xs.split_on '/').drop_while (λ x, x ≠ "library" ∧ x ≠ "src"))
+
+def prepare_filename₂ : list string → list string
+| ("library" :: l) := ("core" :: l)
+| ("src" :: l)     := l
+| _                := []
+
+def filename : string → string :=
+"/".intercalate ∘ prepare_filename₂ ∘ prepare_filename₁
+
+def topic : string → string :=
+list.head ∘ prepare_filename₂ ∘ prepare_filename₁
+
+structure item :=
+(file : string)
+(line : string)
+(name : string)
+(topic : string)
+
+structure node extends item.
+
+namespace node
+
+def jsonify : node → string :=
+λ n,
+"{ \"name\" : \"" ++ n.name ++ "\",\n" ++
+  "\"topic\" : \"" ++ n.topic ++ "\",\n" ++
+  "\"file\" : \"" ++ n.file ++ "\",\n" ++
+  "\"line\" : " ++ n.line ++ " }\n"
```

Comment:
There's now an API for building json, right?

Reply:
Yes, that's right. Feel free to hack on this. I mostly put up this PR so that we don't loose the script.